crates/kira/src/backend/cpal/desktop.rs

```diff
@@ -56,6 +56,44 @@ impl CpalBackend {
 			.pop()
 			.ok()
 	}
+
+	/// Returns the oldest available stream error in the queue.
+	pub fn pop_error(&mut self) -> Option<StreamError> {
+		if let State::Initialized {
+			stream_manager_controller,
+		} = &mut self.state
+		{
+			stream_manager_controller.pop_handled_error()
+		} else {
+			None
+		}
+	}
+
+	/// Returns the number of unhandled stream errors discarded because of overcrowding.
+	/// This increases when cpal produces errors faster than they are polled by kira.
+	pub fn num_unhandled_stream_errors_discarded(&self) -> Option<u64> {
```

Comment:
I don't think we really need to distinguish between the two reasons why an error was discarded. The only thing I can imagine people using this number for is displaying `+30 errors` in the terminal. Can you condense this to one function for the total number of errors discarded for any reason?

Reply:
Sure, I just thought I'd retain as much information as possible. But I can condense this.